### PR TITLE
[TTAHUB-2268] log all unknown errors

### PR DESCRIPTION
**Pull request by GarrettEHill**
## Description of change
Now all unknown errors will log to the Request Errors table unless the env SUPPRESS_ERROR_LOGGING is set to true
This should result in missing/lost ttaProvided or similar to be recoverable more often

## How to test
Trigger an error, look in the RequestErrors DB or the admin UI for the logged data

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-2268

### Every PR

- [x] Meets issue criteria
- [x] JIRA ticket status updated
- [x] Code is meaningfully tested
- [ ] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [ ] API Documentation updated
- [ ] Boundary diagram updated
- [ ] Logical Data Model updated
- [ ] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions
- [ ] UI review complete

### Production Deploy

- [ ] Staging smoke test completed

### After merge/deploy

- [ ] Update JIRA ticket status

### CODE REVIEW

File: src/lib/apiErrorHandler.js [hunk 1]
```diff
@@ -3,48 +3,71 @@ import Sequelize from 'sequelize';
...
- * @param {*} error - error
- * @param {*} logContext - useful data for logging
- */
-async function handleSequelizeError(req, res, error, logContext) {
+async function logRequestError(req, operation, error, logContext) {
+  if (operation !== 'SequelizeError'
+  && process.env.SUPPRESS_ERROR_LOGGING
+  && process.env.SUPPRESS_ERROR_LOGGING.toLowerCase() === 'true') {
+    return 0;
+  }
```
Comment (kryswisnaskas) [inline src/lib/apiErrorHandler.js:11]: Thank you for adding the environment variable. 👍  To make it effective (per env) we still need to add it to the configuration files.

File: src/routes/activityReports/handlers.js [hunk 2]
```diff
@@ -246,7 +246,7 @@ export async function updateLegacyFields(req, res) {
     const savedReport = await createOrUpdate({ imported }, report);
     res.json(savedReport);
   } catch (error) {
-    handleErrors(req, res, error, logContext);
+    await handleErrors(req, res, error, logContext);
   }
```
Comment (kryswisnaskas) [inline src/routes/activityReports/handlers.js:250]: 👍

File: src/lib/apiErrorHandler.js [hunk 3]
```diff
@@ -3,48 +3,71 @@ import Sequelize from 'sequelize';
 import createRequestError from '../services/requestErrors';
 import { auditLogger as logger } from '../logger';
```
Comment (kryswisnaskas) [inline src/lib/apiErrorHandler.js:5]: Can we have the comments back for this new function?

**Review by kryswisnaskas — COMMENTED** [review]
Looks good!

I have a couple of comments in the code.

File: src/lib/apiErrorHandler.js (hunk 1 above)
Comment (GarrettEHill) [reply src/lib/apiErrorHandler.js:11]: which config file are you referring to

Comment (GarrettEHill) [reply src/lib/apiErrorHandler.js:22]: I think i fixed this

**Comment by kryswisnaskas** [comment]
Thanks for adding the env variable. I am assuming Circle CI has it also? It looks like we need it in prod rather quickly.  It would still be good to have separate settings for the non-prod environments. We could handle that change in a separate PR if necessary.

Oh, and one more change. The var needs to be added to `config.yml` as well.

**Review by kryswisnaskas — APPROVED** [review]
👍

**Pull request merged**